deps/wazuh_testing/wazuh_testing/data_visualization_module/dash_app/data_loader.py

```diff
@@ -0,0 +1,141 @@
+        params (Tuple[Any, ...]): possible parameters for the database query.
+
+    Returns:
+        df (pd.DataFrame): Dataframe with the result of the query.
+    """
+    conn = sqlite3.connect(database_file)
+    df = pd.read_sql_query(query, conn, params=params)
+    conn.close()
+    return df
+
+
+def extract_config_parameters(config: Dict[str, Any]) -> Tuple[str, str, List[str]]:
+    """Function that extracts the necessary parameters for database queries from the YAML configuration file.
+
+    Args:
+        config (Dict[str, Any]): dictionary containing the information from the YAML file.
+
+    Returns:
+        component (str): component on which the visualization is to be performed.
+        processes (List[str]): process or processes found in the process_name column of the CSV.
+        columns_to_avoid (List[str]): columns that you do not want to load in the visualization.
+        process_name (str): name of the main process to be displayed.
+    """
+    component = config.get('Component', [])[0]
+
+    if component not in ["manager", "indexer", "dashboard", "agent"]:
+        raise ValueError("The component specified in YAML file is invalid")
+
+    process_name = list(config.get('Processes', {}).keys())[0]
+    processes = config.get('Processes', {}).get(process_name, [])
+    columns_to_avoid = config.get('Columns_to_avoid', [])
+
+    return component, processes, columns_to_avoid, process_name
+
+
+def load_initial_data_from_db(config: Dict[str, Any]) -> Tuple[List[str], List[str], Dict[str, str]]:
+    """Function that loads data from the database for initial display of it in the application.
+
+    Args:
+        config (Dict[str, Any]): dictionary containing the information from the YAML file.
+
+    Returns:
+        processes (List[str]): list of processes to display.
+        column_names (List[str]): columns in the CSV file to be displayed.
+        commit_values (Dict[str, str]): dictionary containing the commit value of each file along with its version.
+    """
+    component, processes, columns_to_avoid, _ = extract_config_parameters(config)
+
+    query = "SELECT filename, file_content FROM file_storage WHERE component = ?"
+    files = query_db(query, (component,))
+
+    if files.empty:
+        print("No files found for the specified component.")
+        return [], [], {}
+
+    column_names = []
+    commit_values = {}
+
+    file_content = io.BytesIO(files.iloc[0]['file_content'])
+    df = pd.read_csv(file_content)
+    column_names = [col for col in df.columns if col not in columns_to_avoid]
+
+    for _, row in files.iterrows():
+        file_content = io.BytesIO(row['file_content'])
+        df = pd.read_csv(file_content, usecols=['Version', 'Commit'])
+        version = df['Version'].iloc[0]
+        commit = df['Commit'].iloc[0]
+
+        if commit not in commit_values:
+            commit_values[commit] = []
+        commit_values[commit].append(version)
+
```

Comment:
Instead of storing the full content of the CSV in the database, we are required to design and create a database structure that suits our requirements. This approach does not have the the required performance